src/main/python/deep_qa/layers/noisy_or.py

```diff
@@ -0,0 +1,89 @@
+from keras import backend as K
+from keras.layers import Layer
+from keras.constraints import Constraint
+from keras.regularizers import L1L2Regularizer
+from overrides import overrides
+
+class BetweenZeroAndOne(Constraint):
+    """
+    Constrains the weights to be between zero and one
+    """
+    def __call__(self, p):
+        # Clip values less than or equal to zero to epsilon
+        p *= K.cast(p >= 0., K.floatx())
+        leaky_zeros_mask = K.epsilon() * K.cast(K.equal(p, K.zeros_like(p)), 'float32')
+        p = p + leaky_zeros_mask
+        # Clip values greater to 1 to 1
+        p *= K.cast(p <= 1., K.floatx())
+        leaky_ones_mask = K.cast(K.equal(p, K.zeros_like(p)), K.floatx())
+        p += leaky_ones_mask
+        return p
+
+
+class NoisyOr(Layer):
+    """
+    Calculate the noisy-or probability across a given tensor axis based on
+    noisy-or equation:
+        p(x) = 1 - Prod_(i=1:N)[1 - q * p(x|y_n)]
+    where q is the noise parameter
+
+    Parameters
```

Comment:
I worked on this -- an am not sure it's exactly what you wanted, but the docstring looks pretty consistent with some other classes (they're not all consistent with each other).  I changed the variable inputs to the call function, essentially naming the input to try to better match the documentation as well as some of the other classes.  previously I just had 'x' and can go back to that if you want -- just let me know